cmake/compiler_config.cmake

```diff
@@ -131,9 +131,10 @@ function(mayaHydra_compile_config TARGET)
         )
     endif()
 
-    # Remove annoying TBB warnings.
+    # Common definitions
     target_compile_definitions(${TARGET}
         PRIVATE
-            TBB_SUPPRESS_DEPRECATED_MESSAGES
+            TBB_SUPPRESS_DEPRECATED_MESSAGES # Remove TBB deprecation warnings
+            BOOST_ALL_NO_LIB # Avoid Boost autolinking libraries
```

Comment:
Thanks for doing this!  Can you check all the places where we use BOOST_DEBUG_PYTHON and BOOST_LINKING_PYTHON?  I believe we can remove all these, as there is no Boost Python our own mayaHydra code, only in code we include from OpenUSD.

Reply:
So it looks like Python is still being linked despite this. I've tried removing the BOOST_*_PYTHON definitions, keeping TBB_USE_DEBUG and moving it to the common definitions, and making the common definitions PUBLIC, but despite this it's still trying to link Python. The minimal setup I've found is to just use BOOST_DEBUG_PYTHON in the common definitions, and when that's there, everything else can be removed and things compile again.

Reply:
Update : I think I might've had something wrong in my setup back then? I just retried it, but instead got some other errors that will no longer happen once https://github.com/Autodesk/maya-hydra/pull/37 is merged. I tried building using those changes and removing the remaining BOOST_DEBUG_PYTHON, and it worked. So once the other PR is merged I think we should be able to get rid of the remaining BOOST_DEBUG_PYTHON.